andrepereiradasilva/joomla-cms
# Add core field to extension (for core extensions)

### Summary of Changes

This adds a new column to extensions table named `core`.
The goal of this column is to mark core extenisons as so, this way they cannot be uninstalled, but, if not `protected`they can be enabled/disabled.

Still work in progress:

- [x] Add core field to `#__extensions` table in installation sqls
- [x] Mark core field values in installation sqls
- [ ] Unprotect non essential extensions in installation sqls
- [x] Add core field in update sqls
- [x] Mark core field values in updates sqls
- [x] Unprotect non essential extensions in update sqls
- [x] Add core column  in extension manage view
- [x] Add core filter in extension manage view
- [ ] Prevent core extensions from being uninstalled

### Testing Instructions

DO NOT TEST WHILE NOT FINISHED.

Two test needed: update, install

- Update
  - Apply patch
  - Run the update query manually 
  - Check in extensions manage if protected extensions are protected and core extensions are marked as so
  - Check you cannot unninstall a core extension but you can disable it if it's not locked.

- Install
  - Apply patch
  - Delete configuration.php and install joomla as usual
  - Check in extensions manage if protected extensions are protected and core extensions are marked as so
  - Check you cannot unninstall a core extension but you can disable it if it's not locked.

### Documentation Changes Required

None.
